Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -706,11 +707,28 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     postAsyncNotification(_WKWebExtensionContextPermissionMatchPatternsWereDeniedNotification, addedMatchPatterns);
 }
 
-void WebExtensionContext::postAsyncNotification(NSNotificationName notificationName, PermissionsSet& permissions)
+void WebExtensionContext::permissionsDidChange(const PermissionsSet& changedPermissions)
+{
+    if (!isLoaded())
+        return;
+
+    if (changedPermissions.contains(_WKWebExtensionPermissionClipboardWrite)) {
+        bool granted = hasPermission(_WKWebExtensionPermissionClipboardWrite);
+
+        enumerateExtensionPages([&](auto& page) {
+            page.preferences().setJavaScriptCanAccessClipboard(granted);
+            return false;
```

Comment:
Does this mean that we will only do this on one page? It appears that this `false` will be put into `shouldContinue`, which makes us return after only one iteration.